### DISCUSSION TITLE
128x64LCD 128X64 I2C ST7567S COG LCD module from Alliexpress

### DISCUSSION THREAD

**Discussion by KR3C1K** [olikraus/u8g2#2307] [Q&A]
Hello,
I am looking for a way to run a 128X64 I2C ST7567S COG LCD display, bought from Alliexpress with the help of U8G2 library under Arduino.
I tested many configurations, but the only thing that worked was "U8G2_ST7567_HEM6432_F_HW_I2C u8g2(U8G2_R0, /* reset=*/ U8X8_PIN_NONE)".
The problem is that it displays a really small logo image.
I add a link to this display [http://tiny.cc/hlqfvz.](url)
For any help, thank you.....

**Comment by olikraus** [comment]
Please check the other constructors, like U8G2_ST7567_JLX12864, U8G2_ST7567_ENH_DG128064, U8G2_ST7567_ENH_DG128064I and U8G2_ST7567_OS12864
In all cases it might required to change the contrast (setContrast()). Maybe use the contrast test ino for testing (it will cycle through the contrast).

**Comment by KR3C1K** [comment]
Thanks for the hint.
However, it turned out that "U8G2_ST7567_ENH_DG128064I_F_SW_I2C u8g2(U8G2_R2, SCL, SDA);" works.
I had to add in "void setup()" the I2C address "u8g2.setI2CAddress(0x3F * 2);" for this LCD before "u8g2.begin();".
I did not pay attention to this before.
